### PR TITLE
fix: remove type inferences from components

### PR DESCRIPTION
**Pull request by adrianthedev**
# Description

Fixes https://github.com/avo-hq/avo/issues/3192

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by rickychilcott** [comment]
For my error (following https://docs.avohq.io/3.0/guides/rest-api-integration.html), I need to make this change to make it pass. I'm happy to make a separate PR, but it feels like it should be together. 

FYI`_Any` does not include `nil` so you _may_ need to do the same to fix https://github.com/avo-hq/avo/issues/3192 for @xeron.

```ruby
class Avo::Views::ResourceEditComponent
   ...
   prop :record, _Nilable(_Any)
   ...
end
```

**Comment by xeron** [comment]
> FYI`_Any` does not include `nil` so you _may_ need to do the same to fix #3192 for @xeron.

I'm not setting `current_user` to `nil`, I'm passing a `Struct`, only `name` attribute is `nil`.

File: app/components/avo/sidebar_profile_component.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class Avo::SidebarProfileComponent < Avo::BaseComponent
-  prop :user, _Nilable(ActiveRecord::Base)
+  prop :user, _Any
```
Comment (Paul-Bob) [inline app/components/avo/sidebar_profile_component.rb:4]:
```suggestion
  prop :user, _Nilable(_Any)
```

**Review by Paul-Bob — APPROVED** [review]

**Comment by rickychilcott** [comment]
We discovered a second error today. So far the following would fix it for us:

```ruby
class Avo::PanelComponent < Avo::BaseComponent
  ...
  prop :name, _Nilable(Union(_String, _Integer)) do |value|
    value || @args&.dig(:title)
  end
  ...
end

class Avo::Views::ResourceEditComponent < Avo::ResourceComponent
  ...
  prop :resource, _Nilable(_Any)
  ...
end
```

**Comment by rickychilcott** [comment]
Lies... This is what I had to do (so far):

```ruby
# in config/initializers/avo.rb

module OverrideAvoPanelComponent
  extend ActiveSupport::Concern

  included do
    prop :name, _Nilable(_Union(_String, _Integer)) do |value|
      value || @args&.dig(:title)
    end
  end
end

module OverrideAvoPanelNameComponent
  extend ActiveSupport::Concern

  included do
    prop :name, _Nilable(_Union(_String, _Integer))
  end
end

module OverrideAvoResourceEditComponent
  extend ActiveSupport::Concern

  included do
    prop :resource, _Nilable(_Any)
  end
end

module OverrideAvoFieldBadgeViewerComponent
  extend ActiveSupport::Concern

  included do
    prop :value, _Union(_String, _Symbol)
  end
end

Rails.configuration.to_prepare do
  Avo::PanelComponent.include(OverrideAvoPanelComponent)
  Avo::PanelNameComponent.include(OverrideAvoPanelNameComponent)
  Avo::Fields::Common::BadgeViewerComponent.include(OverrideAvoFieldBadgeViewerComponent)
  Avo::Views::ResourceEditComponent.include(OverrideAvoResourceEditComponent)
end
```

**Comment by adrianthedev** [comment]
Thanks @rickychilcott for brainstorming this with us.
I don't quite get why cast the resource as any here `prop :resource, _Nilable(_Any)`?
I mean... we are expecting a resource to be given otherwise it will most likely crash.

**Pull request merged**